node/pkg/terra/watcher.go

```diff
@@ -83,12 +92,24 @@ func NewWatcher(
 	contract string,
 	lockEvents chan *common.MessagePublication,
 	setEvents chan *common.GuardianSet,
-	obsvReqC chan *gossipv1.ObservationRequest) *Watcher {
-	return &Watcher{urlWS: urlWS, urlLCD: urlLCD, contract: contract, msgChan: lockEvents, setChan: setEvents, obsvReqC: obsvReqC}
+	obsvReqC chan *gossipv1.ObservationRequest,
+	readiness readiness.Component,
+	chainID vaa.ChainID) *Watcher {
+
+	// CosmWasm 1.0.0
+	contractAddressFilterKey := "execute._contract_address"
+	contractAddressLogKey := "_contract_address"
+	if chainID == vaa.ChainIDTerra {
+		// CosmWasm <1.0.0
+		contractAddressFilterKey = "execute_contract.contract_address"
+		contractAddressLogKey = "contract_address"
+	}
```

Comment:
These bits are utter magic to me.

![no idea dog](https://i.imgflip.com/6jy9hs.jpg)

Reply:
lol that's what was in the watcher before (for terra classic) but CosmWasm 1.0.0 has different fields for the contract address on logs and filters https://github.com/terra-money/core/blob/main/WASM_MIGRATE.md#execute